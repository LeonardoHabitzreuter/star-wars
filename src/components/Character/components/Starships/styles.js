import styled from 'styled-components'

export const StarhipsList = styled.article`
  box-sizing: border-box;
  display: flex;
  flex-wrap: wrap;
  margin: 16px auto;
  max-width: 1324px;
`
