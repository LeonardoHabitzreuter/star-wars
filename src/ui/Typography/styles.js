import styled, { css } from 'styled-components'

const commonStyle = ({ bold, color, align }) => css`
  color: ${color};
  margin: 0;
  font-weight: ${bold ? 'bold' : 'normal'};
  line-height: 1.5;
  text-align: ${align};
`

const H1 = styled.h1`
  font-size: 42px;
  ${commonStyle}
`

const H2 = styled.h2`
  font-size: 36px;
  ${commonStyle}
`

const H3 = styled.h3`
  font-size: 30px;
  ${commonStyle}
`

const H4 = styled.h4`
  font-size: 24px;
  ${commonStyle}
`

const H5 = styled.h5`
  font-size: 22px;
  ${commonStyle}
`

const H6 = styled.h6`
  font-size: 18px;
  ${commonStyle}
`

const Subtitle = styled.span`
  font-size: 16px;
  ${commonStyle}
`

const Text = styled.p`
  font-size: 16px;
  ${commonStyle}
`

const Body = styled.span`
  font-size: 14px;
  ${commonStyle}
`

export const button = css`
  font-size: 18px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.33;
  letter-spacing: normal;
  text-align: center;
`

export default { H1, H2, H3, H4, H5, H6, Subtitle, Text, Body }
