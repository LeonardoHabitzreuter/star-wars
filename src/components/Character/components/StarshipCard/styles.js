import Card from '/ui/Card'
import styled, { css } from 'styled-components'
import { Typography } from '/ui'
import { phone, tablet } from '/styles/theme'

export const StyledStarship = styled(Card)`
  ${({ hide }) => hide && 'display: none;'}
  box-sizing: border-box;
  height: auto;
  width: 100%;
  --cols: 3;
  max-width: calc(100% / var(--cols) - 20px);
  margin: 0 10px;
  margin-bottom: 16px;
  box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.14);

  ${tablet(css`
    --cols: 2;
  `)}

  ${phone(css`
    --cols: 1;
    max-width: 100%;
    margin: 10px 0;
  `)}
`

export const Row = styled.div`
  width: 100%;
  margin: 5px 0;
  display: flex;
  justify-content: space-between;
`

export const LeftCol = styled(Typography)`
  width: 61%;
`

export const RightCol = styled(Typography)`
  width: 39%;
`

export const Title = styled(Typography)`
  display: inline-flex;
`
