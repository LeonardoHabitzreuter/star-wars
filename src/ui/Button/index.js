import React from 'react'
import PropTypes from 'prop-types'
import { Button, Link } from './styles'

const LinkOrButton = ({ href, children, ...rest }) => (
  !href
    ? <Button {...rest}>{children}</Button>
    : (
      <Link to={href} {...rest}>
        <Button {...rest}>{children}</Button>
      </Link>
    )
)

LinkOrButton.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  size: PropTypes.oneOf(['normal', 'small', 'full']),
  modifier: PropTypes.oneOf(['primary', 'secondary']),
  href: PropTypes.string
}

LinkOrButton.defaultProps = {
  type: 'button',
  size: 'normal',
  modifier: 'primary'
}

export default LinkOrButton
