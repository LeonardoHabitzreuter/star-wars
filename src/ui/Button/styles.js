import * as R from 'ramda'
import styled, { css } from 'styled-components'
import { darken, lighten } from 'polished'
import { Link as RouterLink } from 'react-router-dom'
import { phone } from '/styles/theme'
import { button } from '/ui/Typography/styles'

const fullSizeProps = css`
  padding: 8px 20px;
  width: 100%;
`

const smallSizeProps = css`
  padding: 4px 12px;

  ${phone(css`
    ${fullSizeProps}
  `)}
`

const defaultSize = css`
  padding: 8px 20px;

  ${phone(css`
    ${fullSizeProps}
  `)}
`

const primaryCss = ({ theme: { colors } }) => css`
  color: white;
  border: none;
  background-color: ${colors.primary};
  box-shadow: 1px 5px 5px ${colors.primary};
  &:hover {
    background-color: ${lighten(0.1, colors.primary)}
  }

  &:active {
    background-color: ${darken(0.1, colors.primary)}
  }
`

const secondaryCss = ({ theme: { colors } }) => css`
  color: ${colors.secondary};
  border: solid 1px ${darken(0.2, colors.primary)};
  background-color: ${lighten(0.25, colors.primary)};
  &:active {
    background-color: ${lighten(0.1, colors.primary)}
  }
`

const modifierEq = R.propEq('modifier')
const getColorProps = R.cond([
  [modifierEq('primary'), primaryCss],
  [modifierEq('secondary'), secondaryCss],
  [R.T, primaryCss]
])

const sizeEq = R.propEq('size')
const getSizeProps = R.cond([
  [sizeEq('small'), R.always(smallSizeProps)],
  [sizeEq('full'), R.always(fullSizeProps)],
  [R.T, R.always(defaultSize)]
])

export const Button = styled.button`
  border-radius: 2px;
  outline: none;
  cursor: pointer;

  ${getSizeProps}
  ${getColorProps}
  ${button}
`

export const Link = styled(RouterLink)`
  ${getSizeProps};
  padding: 0;
`
