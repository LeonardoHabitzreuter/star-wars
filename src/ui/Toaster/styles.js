import styled from 'styled-components'
import { colors } from '/styles/theme'

export const StyledWrapper = styled.div`
  .Toastify__toast-container {
  }

  .Toastify__toast {
    border-radius: 4px;
    padding: 0;
    padding: 24px;
    box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.1);
    font-weight: 500;
    font-size: 18px;
  }

  .Toastify__toast--error {
    background: ${colors.danger};
  }

  .Toastify__toast--success {
    background: #69c724;
  }

  .Toastify__toast--warning {
    background: yellow;
  }
`
