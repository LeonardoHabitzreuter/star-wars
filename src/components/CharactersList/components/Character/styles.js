import Card from '/ui/Card'
import styled, { css } from 'styled-components'
import { Typography } from '/ui'

export const StyledCharacter = styled(Card)`
  ${({ hide }) => hide && 'display: none;'}
  box-sizing: border-box;
  height: 100px;
  width: 100%;
  --cols: 3;
  max-width: calc(100% / var(--cols) - 20px);
  margin: 0 10px;
  margin-bottom: 16px;
  box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.14);

  ${({ theme: { sizes } }) => css`
    @media (max-width: ${sizes.tablet}) {
      --cols: 2;
    }

    @media (max-width: ${sizes.phone}) {
      --cols: 1;
      max-width: 100%;
      margin: 10px 0;
    }
  `}
`

export const Row = styled.div`
  width: 100%;
  margin: 5px 0;
  display: flex;
  justify-content: space-between;
`

export const Characteristic = styled(Typography)`
  width: 50%;
  text-align: center;
`
