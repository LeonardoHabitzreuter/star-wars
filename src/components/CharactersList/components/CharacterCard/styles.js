import Card from '/ui/Card'
import styled, { css } from 'styled-components'
import { Button, Typography } from '/ui'
import { phone, tablet } from '/styles/theme'

export const StyledCharacter = styled(Card)`
  ${({ hide }) => hide && 'display: none;'}
  box-sizing: border-box;
  height: 150px;
  width: 100%;
  --cols: 3;
  max-width: calc(100% / var(--cols) - 20px);
  margin: 0 10px;
  margin-bottom: 16px;
  box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.14);

  ${tablet(css`
    --cols: 2;
  `)}

  ${phone(css`
    --cols: 1;
    max-width: 100%;
    margin: 10px 0;
  `)}
`

export const Row = styled.div`
  width: 100%;
  margin: 5px 0;
  display: flex;
  justify-content: space-between;
`

export const Characteristic = styled(Typography)`
  width: 50%;
`

export const DetailsButton = styled(Button)`
  align-self:flex-end;

  ${phone(css`
    align-self: center;
  `)}
`
