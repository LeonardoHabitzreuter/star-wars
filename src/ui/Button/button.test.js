import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import * as theme from '/styles/theme'
import React from 'react'
import Button from '/ui/Button'
import { render } from '@testing-library/react'

describe('Button component', () => {
  test('should render properly', () => {
    const { container } = render(
      <Button theme={theme}>Test</Button>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
