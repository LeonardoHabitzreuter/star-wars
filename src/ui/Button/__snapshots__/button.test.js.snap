// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render properly 1`] = `
.c0 {
  border-radius: 2px;
  outline: none;
  cursor: pointer;
  padding: 8px 20px;
  color: white;
  border: none;
  background-color: #141414;
  box-shadow: 1px 5px 5px #141414;
  font-size: 18px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.33;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: center;
}

.c0:hover {
  background-color: #2e2e2e;
}

.c0:active {
  background-color: #000;
}

@media (max-width:414px) {
  .c0 {
    padding: 8px 20px;
    width: 100%;
  }
}

<button
  class="c0"
  type="button"
>
  Test
</button>
`;
